### ISSUE TITLE
[immutable-data] Arrays from `Object.entries` and the like are not treated as new arrays being created

### ISSUE THREAD

**Issue by RebeccaStevens** [eslint-functional/eslint-plugin-functional#773]
## Bug Report

This errors with `ignoreImmediateMutation` set.

```ts
Object.entries(foo).sort(...);
```

### Expected behavior

No error.

### Actual behavior

Modifying an array is not allowed.
